libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c

```diff
@@ -1905,71 +1962,43 @@ static OTA_FileContext_t * prvParseJobDoc( const char * pcJSON,
         if( eErr == eOTA_JobParseErr_None )
         {
             /* If the job is in self test mode, don't start an
-              * OTA update but instead do the following:
-              *
-              * If the firmware that performed the update was older
-              * than the currently running firmware, set the image
-              * state to "Testing." This is the success path.
-              *
-              * If it's the same or newer, reject the job since
-              * either the firmware wasn't accepted during self
-              * test or an incorrect image was sent by the OTA
-              * operator.
-              */
+             * OTA update but instead do the following:
+             *
+             * If the firmware that performed the update was older
+             * than the currently running firmware, set the image
+             * state to "Testing." This is the success path.
+             *
+             * If it's the same or newer, reject the job since
+             * either the firmware wasn't accepted during self
+             * test or an incorrect image was sent by the OTA
+             * operator.
+             */
             if( C->xIsInSelfTest == ( bool_t ) pdTRUE )
             {
                 OTA_LOG_L1( "[%s] In self test mode.\r\n", OTA_METHOD_NAME );
 
-                /* Only check for versions if the target is self */
-                if( xOTA_Agent.ulServerFileID == 0 )
+                /* Validate version of the update received.*/
+                xErrVersionCheck = prvValidateUpdateVersion( C );
+
+                if( otaconfigAllowDowngrade || ( xErrVersionCheck == kOTA_Err_None ) )
```

Comment:
Does this bypass all checks (including invalid image) if otaconfigAllowDowngrade  is set to true? Shouldn't it still check for invalid client token?

Reply:
We are only bypassing the version check if the config option is enabled ( we will keep it disabled by default) . This does not bypass any other checks and still logs if downgrade and same version update is performed.

Reply:
Prasad will add a test that a job document with invalid client token will be rejected in both modes (downgrade allowed and not allowed).